source/Octopus.Tentacle.Tests.Integration/Support/ExtensionMethods/FluentAssertionExtensionMethods.cs

```diff
@@ -1,15 +1,36 @@
 ﻿using System;
-using System.Threading.Tasks;
 using FluentAssertions;
 using FluentAssertions.Primitives;
+using Halibut;
+using Halibut.Exceptions;
 
 namespace Octopus.Tentacle.Tests.Integration.Support.ExtensionMethods
 {
     public static class FluentAssertionExtensionMethods
     {
-        public static AndConstraint<ObjectAssertions> BeTaskOrOperationCancelledException(this ObjectAssertions should)
+        public static AndConstraint<ObjectAssertions> BeRequestCancelledException(this ObjectAssertions should, RpcCallStage rpcCallStage)
         {
-            return should.Match(x => x is TaskCanceledException || x is OperationCanceledException);
+            if (rpcCallStage == RpcCallStage.Connecting)
+            {
+                return should.Match(x => x is ConnectingRequestCancelledException || IsHalibutWrappedConnectingRequestCancelledException(x as HalibutClientException));
+            }
+
+            return should.Match(x => x is TransferringRequestCancelledException || IsHalibutWrappedTransferringRequestCancelledException(x as HalibutClientException));
+        }
+
+        static bool IsHalibutWrappedConnectingRequestCancelledException(HalibutClientException? halibutClientException)
+        {
+            return halibutClientException != null && halibutClientException.Message.Contains("The Request was cancelled while Connecting");
```

Comment:
Like before, if possible, is there any other property that contains maybe the type as a string? Something else to compare against other than the message 🙂